src/wccp2.cc

```diff
@@ -2024,10 +2033,16 @@ wccp2AssignBuckets(void *)
                                     tmp_rtr,
                                     &wccp_packet,
                                     offset);
+                    // XXX: setup fd_table[] properly so we can use FD_WRITE_METHOD()
+                    fd_table[theWccp2Connection->fd].bytesWritten(offset);
                 } else {
-                    if (send(theWccp2Connection, &wccp_packet, offset, 0) < static_cast<int>(offset)) {
+                    const auto x = send(theWccp2Connection, &wccp_packet, offset, 0);
+                    if (x < static_cast<int>(offset)) {
                         int xerrno = errno;
                         debugs(80, 2, "ERROR: failed to send WCCPv2 HERE_I_AM packet to " << tmp_rtr << " : " << xstrerr(xerrno));
+                    } else {
+                        // XXX: setup fd_table[] properly so we can use FD_WRITE_METHOD()
+                        fd_table[theWccp2Connection->fd].bytesWritten(x);
```

Comment:
> PR description: Move the FD read/write byte counting to the class fde read/write operators ... This fixes ... under-counting of UDP protocols

This PR does not move UDP byte accounting to fde class methods. Instead, it fixes UDP undercounting by adding missing accounting calls to UDP handling code. The current plan is to switch UDP handling code to fde accounting as well, but that has not happened yet. Please adjust the PR description accordingly.

While you are fixing this, please replace "operators" with a more accurate "methods" or "functions".